apps/www/_blog/2024-01-26-flutter-figma-clone.mdx

```diff
@@ -0,0 +1,861 @@
+---
+title: Create a Figma Clone app with Flutter and Supabase Realtime
+description: 'A tutorial on how to build a collaborative design app like Figma using Flutter and Supabase Realtime.'
+author: tyler_shukert
+image: flutter-figma-clone/og.png
+thumb: flutter-figma-clone/thumb.png
+categories:
+  - developers
+tags:
+  - flutter
+  - realtime
+date: '2024-01-26'
+toc_depth: 3
+---
+
+Supabase has a low latency real-time communication feature called [Broadcast](https://supabase.com/docs/guides/realtime/broadcast). With it, you can have your clients communicate with other clients with low latencies. This is useful for creating apps with connected experiences. Flutter has a [CustomPainter](https://api.flutter.dev/flutter/rendering/CustomPainter-class.html) class, which allows developers to interact with the low-level canvas API allowing us to render virtually anything on the app. Combining these two tools allows us to create interactive apps.
+
+In this article, I am combining the Supabase Realtime Broadcast with Flutter’s `CustomPainter` to create a collaborative design board app like Figma.
+
+You can find the full code example [here](https://github.com/supabase/supabase/tree/master/examples/realtime/flutter-figma-clone).
+
+## Overview of the Figma clone app
+
+We are building an interactive design canvas app where multiple users can collaborate in real time. We will add the following features to the app:
+
+- Draw shapes such as circles or rectangles
+- Move those shapes around
+- Sync the cursor position and the design objects with other clients in real-time
+- Persist the state of the canvas in a Postgres database
+
+<video width="99%" loop muted playsInline controls={true}>
+  <source
+    src="https://obuldanrptloktxcffvn.supabase.co/storage/v1/object/public/videos/marketing/blog/flutter-figma-clone/flutter-figma-clone.mp4"
+    type="video/mp4"
+  />
+</video>
+
+https://obuldanrptloktxcffvn.supabase.co/storage/v1/object/public/videos/marketing/blog/flutter-figma-clone/flutter-figma-clone.mp4
```

Comment:
this is probably a typo, right? @dshukertjr

Reply:
Ah, shoot. Sorry for that. I have included a fix here

https://github.com/supabase/supabase/pull/20782